### ISSUE TITLE
Error: amqp: sender settlement mode "unsettled" requested, received "mixed" from server

### ISSUE THREAD

**Issue by jonnylangefeld** [Azure/azure-event-hubs-go#115]
I just instantiated a regular eventhub with 10 partitions. I am trying to send a simple byte array, but i get below error message. I see that the `amqp.ModeUnsettled` is [hard coded](https://github.com/Azure/azure-event-hubs-go/blob/eb729b0d2cd97ad9a628e7ba3c99c679c8cf0a0b/sender.go#L251), so I have no influence on that. Do I have to do something different on EventHub Creation?

    hub, err = eventhub.NewHubFromConnectionString(connection)
    err = hub.Send(ctx, eventhub.NewEvent(eventByte))

### Expected Behavior
message is sent to Eventhub

### Actual Behavior
I get the error

    amqp: sender settlement mode "unsettled" requested, received "mixed" from server

### Environment
- OS: Mac OS Mojave
- Go version: `go1.12.4 darwin/amd64`
- Version of Library: `v2@v2.0.0`

**Comment by devigned** [comment]
Can you run this with `DEBUG_LEVEL=3 go run ${your_app_path} -tags debug`, and post the output?

**Comment by devigned** [comment]
Here's what I would expect for sending to a partitioned Event Hub targeting a single partition to send messages:

```
08:37:03.721290 TX: Attach{Name: eQn5vPnhP_xjaJEhKpQ4Pn0ITwWS2KukzFOdDmfI78nUEsAmC9aiuQ, Handle: 0, Role: Sender, SenderSettleMode: unsettled, ReceiverSettleMode: <nil>, Source: <nil>, Target: source{Address: goehtestbwcli2-tagf22t4/Partitions/0, Durable: 0, ExpiryPolicy: , Timeout: 0, Dynamic: false, DynamicNodeProperties: map[], Capabilities: []}, Unsettled: map[], IncompleteUnsettled: false, InitialDeliveryCount: 0, MaxMessageSize: 0, OfferedCapabilities: [], DesiredCapabilities: [], Properties: map[]}
08:37:03.999380 RX(Session): Attach{Name: eQn5vPnhP_xjaJEhKpQ4Pn0ITwWS2KukzFOdDmfI78nUEsAmC9aiuQ, Handle: 0, Role: Receiver, SenderSettleMode: unsettled, ReceiverSettleMode: first, Source: <nil>, Target: source{Address: goehtestbwcli2-tagf22t4/Partitions/0, Durable: 0, ExpiryPolicy: session-end, Timeout: 0, Dynamic: false, DynamicNodeProperties: map[], Capabilities: []}, Unsettled: map[], IncompleteUnsettled: false, InitialDeliveryCount: 0, MaxMessageSize: 1046528, OfferedCapabilities: [], DesiredCapabilities: [], Properties: map[]}
08:37:03.999457 RX(Session): Flow{NextIncomingID: 0, IncomingWindow: 5000, NextOutgoingID: 1, OutgoingWindow: 100, Handle: 0, DeliveryCount: 0, LinkCredit: 1000, Available: 0, Drain: false, Echo: false, Properties: map[]}
08:37:03.999587 RX: Attach{Name: eQn5vPnhP_xjaJEhKpQ4Pn0ITwWS2KukzFOdDmfI78nUEsAmC9aiuQ, Handle: 0, Role: Receiver, SenderSettleMode: unsettled, ReceiverSettleMode: first, Source: <nil>, Target: source{Address: goehtestbwcli2-tagf22t4/Partitions/0, Durable: 0, ExpiryPolicy: session-end, Timeout: 0, Dynamic: false, DynamicNodeProperties: map[], Capabilities: []}, Unsettled: map[], IncompleteUnsettled: false, InitialDeliveryCount: 0, MaxMessageSize: 1046528, OfferedCapabilities: [], DesiredCapabilities: [], Properties: map[]}
08:37:03.999772 RX: Flow{NextIncomingID: 0, IncomingWindow: 5000, NextOutgoingID: 1, OutgoingWindow: 100, Handle: 0, DeliveryCount: 0, LinkCredit: 1000, Available: 0, Drain: false, Echo: false, Properties: map[]}
08:37:03.999878 TX(link): Transfer{Handle: 0, DeliveryID: 1, DeliveryTag: "\x00\x00\x00\x00\x00\x00\x00\x00", MessageFormat: 0, Settled: false, More: false, ReceiverSettleMode: <nil>, State: <nil>, Resume: false, Aborted: false, Batchable: false, Payload [size]: 46}
08:37:03.999916 TX(Session): Transfer{Handle: 0, DeliveryID: 1, DeliveryTag: "\x00\x00\x00\x00\x00\x00\x00\x00", MessageFormat: 0, Settled: false, More: false, ReceiverSettleMode: <nil>, State: <nil>, Resume: false, Aborted: false, Batchable: false, Payload [size]: 46}
```

**Comment by jonnylangefeld** [comment]
I tried it in my actual app and it didn't work there, so I made a simple file and stripped it down to the essentials, but also there none of the debug output was printed. Here is the code I used in a `main.go` file:

```
package main

import (
	"context"
	"fmt"
	eventhub "github.com/Azure/azure-event-hubs-go/v2"
)

func main() {
	ctx := context.Background()
	hub, _ := eventhub.NewHubFromConnectionString("Endpoint=sb://...")
	err := hub.Send(ctx, eventhub.NewEvent([]byte(`hi`)))
	if err != nil {
		fmt.Printf(err.Error())
	}
}
```

And this is my run command:

    DEBUG_LEVEL=3 go run main.go -tags debug

The only output is still:

    amqp: sender settlement mode "unsettled" requested, received "mixed" from server

**Comment by devigned** [comment]
I'm guessing that `go build` cached some of the underlying package products and didn't recompile with the tag.

```
	-a
		force rebuilding of packages that are already up-to-date.
```

Give it a shot with `-a`.

**Comment by jonnylangefeld** [comment]
Thanks @devigned, that helped a lot! I got the debug output and found out through that, that it was actually my bad, I had to consumers, one for `listen` and one for `send`. I accidentally copied the connection string from the `listen` one. I guess the only thing would be why the error message was misleading. Cause the last line of the debug log actually gave the correct hint that the `send` claim wasn't given.

I'll past the debug log here and leave it up to you if you want to leave the issue open for the correct error message, or if you want to close it. For my problem it is resolved for now :) Thanks again!

```
11:42:59.408091 TX: Begin{RemoteChannel: 0, NextOutgoingID: 0, IncomingWindow: 100, OutgoingWindow: 100, HandleMax: 4294967295, OfferedCapabilities: [], DesiredCapabilities: [], Properties: map[]}
11:42:59.541931 RX: Begin{RemoteChannel: 0, NextOutgoingID: 1, IncomingWindow: 5000, OutgoingWindow: 100, HandleMax: 255, OfferedCapabilities: [], DesiredCapabilities: [], Properties: map[]}
11:42:59.544376 TX: Attach{Name: Y55jLBh3THCTSjk9w6jfgeOOFBYSxUvOg8NvIq2nhDfpR5WZ8eiD7A, Handle: 0, Role: Sender, SenderSettleMode: <nil>, ReceiverSettleMode: <nil>, Source: <nil>, Target: source{Address: $cbs, Durable: 0, ExpiryPolicy: , Timeout: 0, Dynamic: false, DynamicNodeProperties: map[], Capabilities: []}, Unsettled: map[], IncompleteUnsettled: false, InitialDeliveryCount: 0, MaxMessageSize: 0, OfferedCapabilities: [], DesiredCapabilities: [], Properties: map[]}
11:42:59.677739 RX(Session): Attach{Name: Y55jLBh3THCTSjk9w6jfgeOOFBYSxUvOg8NvIq2nhDfpR5WZ8eiD7A, Handle: 0, Role: Receiver, SenderSettleMode: <nil>, ReceiverSettleMode: <nil>, Source: <nil>, Target: source{Address: $cbs, Durable: 0, ExpiryPolicy: session-end, Timeout: 0, Dynamic: false, DynamicNodeProperties: map[], Capabilities: []}, Unsettled: map[], IncompleteUnsettled: false, InitialDeliveryCount: 0, MaxMessageSize: 18446744073709551615, OfferedCapabilities: [], DesiredCapabilities: [], Properties: map[]}
11:42:59.677837 RX(Session): Flow{NextIncomingID: 0, IncomingWindow: 5000, NextOutgoingID: 1, OutgoingWindow: 100, Handle: 0, DeliveryCount: 0, LinkCredit: 100, Available: 0, Drain: false, Echo: false, Properties: map[]}
11:42:59.677840 RX: Attach{Name: Y55jLBh3THCTSjk9w6jfgeOOFBYSxUvOg8NvIq2nhDfpR5WZ8eiD7A, Handle: 0, Role: Receiver, SenderSettleMode: <nil>, ReceiverSettleMode: <nil>, Source: <nil>, Target: source{Address: $cbs, Durable: 0, ExpiryPolicy: session-end, Timeout: 0, Dynamic: false, DynamicNodeProperties: map[], Capabilities: []}, Unsettled: map[], IncompleteUnsettled: false, InitialDeliveryCount: 0, MaxMessageSize: 18446744073709551615, OfferedCapabilities: [], DesiredCapabilities: [], Properties: map[]}
11:42:59.678211 TX: Attach{Name: a-iHMn4zIHCtKCwdx_LBQXMMvWz5YMGz2PhE-5zu0BpKhomBROWdog, Handle: 1, Role: Receiver, SenderSettleMode: <nil>, ReceiverSettleMode: <nil>, Source: source{Address: $cbs, Durable: 0, ExpiryPolicy: , Timeout: 0, Dynamic: false, DynamicNodeProperties: map[], DistributionMode: , Filter: map[], DefaultOutcome: <nil>Outcomes: [], Capabilities: []}, Target: source{Address: $cbs$$cbs-reply-to-63fe2ca2-01ca-4a0f-8332-5ec32bf31837, Durable: 0, ExpiryPolicy: , Timeout: 0, Dynamic: false, DynamicNodeProperties: map[], Capabilities: []}, Unsettled: map[], IncompleteUnsettled: false, InitialDeliveryCount: 0, MaxMessageSize: 0, OfferedCapabilities: [], DesiredCapabilities: [], Properties: map[]}
11:42:59.680125 RX: Flow{NextIncomingID: 0, IncomingWindow: 5000, NextOutgoingID: 1, OutgoingWindow: 100, Handle: 0, DeliveryCount: 0, LinkCredit: 100, Available: 0, Drain: false, Echo: false, Properties: map[]}
11:42:59.927463 RX(Session): Attach{Name: a-iHMn4zIHCtKCwdx_LBQXMMvWz5YMGz2PhE-5zu0BpKhomBROWdog, Handle: 1, Role: Sender, SenderSettleMode: <nil>, ReceiverSettleMode: <nil>, Source: source{Address: $cbs, Durable: 0, ExpiryPolicy: session-end, Timeout: 0, Dynamic: false, DynamicNodeProperties: map[], DistributionMode: , Filter: map[], DefaultOutcome: <nil>Outcomes: [], Capabilities: []}, Target: source{Address: $cbs$$cbs-reply-to-63fe2ca2-01ca-4a0f-8332-5ec32bf31837, Durable: 0, ExpiryPolicy: session-end, Timeout: 0, Dynamic: false, DynamicNodeProperties: map[], Capabilities: []}, Unsettled: map[], IncompleteUnsettled: false, InitialDeliveryCount: 0, MaxMessageSize: 18446744073709551615, OfferedCapabilities: [], DesiredCapabilities: [], Properties: map[]}
11:42:59.927564 RX: Attach{Name: a-iHMn4zIHCtKCwdx_LBQXMMvWz5YMGz2PhE-5zu0BpKhomBROWdog, Handle: 1, Role: Sender, SenderSettleMode: <nil>, ReceiverSettleMode: <nil>, Source: source{Address: $cbs, Durable: 0, ExpiryPolicy: session-end, Timeout: 0, Dynamic: false, DynamicNodeProperties: map[], DistributionMode: , Filter: map[], DefaultOutcome: <nil>Outcomes: [], Capabilities: []}, Target: source{Address: $cbs$$cbs-reply-to-63fe2ca2-01ca-4a0f-8332-5ec32bf31837, Durable: 0, ExpiryPolicy: session-end, Timeout: 0, Dynamic: false, DynamicNodeProperties: map[], Capabilities: []}, Unsettled: map[], IncompleteUnsettled: false, InitialDeliveryCount: 0, MaxMessageSize: 18446744073709551615, OfferedCapabilities: [], DesiredCapabilities: [], Properties: map[]}
11:42:59.928421 TX(link): Transfer{Handle: 0, DeliveryID: 1, DeliveryTag: "\x00\x00\x00\x00\x00\x00\x00\x00", MessageFormat: 0, Settled: false, More: true, ReceiverSettleMode: <nil>, State: <nil>, Resume: false, Aborted: false, Batchable: false, Payload [size]: 446}
11:42:59.928465 TX(link): Transfer{Handle: 0, DeliveryID: <nil>, DeliveryTag: "<nil>", MessageFormat: <nil>, Settled: false, More: false, ReceiverSettleMode: <nil>, State: <nil>, Resume: false, Aborted: false, Batchable: false, Payload [size]: 3}
11:42:59.928632 TX(Session): Transfer{Handle: 0, DeliveryID: 1, DeliveryTag: "\x00\x00\x00\x00\x00\x00\x00\x00", MessageFormat: 0, Settled: false, More: true, ReceiverSettleMode: <nil>, State: <nil>, Resume: false, Aborted: false, Batchable: false, Payload [size]: 446}
11:42:59.928673 TX(Session): Transfer{Handle: 0, DeliveryID: <nil>, DeliveryTag: "<nil>", MessageFormat: <nil>, Settled: false, More: false, ReceiverSettleMode: <nil>, State: <nil>, Resume: false, Aborted: false, Batchable: false, Payload [size]: 3}
11:42:59.929866 TX: Flow{NextIncomingID: <nil>, IncomingWindow: 0, NextOutgoingID: 0, OutgoingWindow: 0, Handle: 1, DeliveryCount: 0, LinkCredit: 1, Available: <nil>, Drain: false, Echo: false, Properties: map[]}
11:42:59.929917 TX(Session): Flow{NextIncomingID: 1, IncomingWindow: 100, NextOutgoingID: 2, OutgoingWindow: 100, Handle: 1, DeliveryCount: 0, LinkCredit: 1, Available: <nil>, Drain: false, Echo: false, Properties: map[]}
11:43:00.070734 RX(Session): Disposition{Role: Receiver, First: 1, Last: <nil>, Settled: true, State: Accepted, Batchable: false}
11:43:00.070790 RX: Disposition{Role: Receiver, First: 1, Last: <nil>, Settled: true, State: Accepted, Batchable: false}
11:43:00.071537 RX(Session): Transfer{Handle: 1, DeliveryID: 0, DeliveryTag: "\x01\x00\x00\x00", MessageFormat: 0, Settled: false, More: false, ReceiverSettleMode: <nil>, State: <nil>, Resume: false, Aborted: false, Batchable: false, Payload [size]: 73}
11:43:00.071603 RX: Transfer{Handle: 1, DeliveryID: 0, DeliveryTag: "\x01\x00\x00\x00", MessageFormat: 0, Settled: false, More: false, ReceiverSettleMode: <nil>, State: <nil>, Resume: false, Aborted: false, Batchable: false, Payload [size]: 73}
11:43:00.071747 TX: Flow{NextIncomingID: <nil>, IncomingWindow: 0, NextOutgoingID: 0, OutgoingWindow: 0, Handle: 1, DeliveryCount: 1, LinkCredit: 1, Available: <nil>, Drain: false, Echo: false, Properties: map[]}
11:43:00.071789 TX: Disposition{Role: Receiver, First: 0, Last: <nil>, Settled: true, State: Accepted, Batchable: false}
11:43:00.071910 TX(Session): Flow{NextIncomingID: 2, IncomingWindow: 100, NextOutgoingID: 2, OutgoingWindow: 100, Handle: 1, DeliveryCount: 1, LinkCredit: 1, Available: <nil>, Drain: false, Echo: false, Properties: map[]}
11:43:00.072132 TX(Session): Detach{Handle: 1, Closed: true, Error: *Error(nil)}
11:43:00.258149 RX(Session): Detach{Handle: 1, Closed: true, Error: *Error(nil)}
11:43:00.258280 TX(Session): Detach{Handle: 0, Closed: true, Error: *Error(nil)}
11:43:00.390970 RX(Session): Detach{Handle: 0, Closed: true, Error: *Error(nil)}
11:43:00.523557 TX: Begin{RemoteChannel: 0, NextOutgoingID: 0, IncomingWindow: 100, OutgoingWindow: 100, HandleMax: 4294967295, OfferedCapabilities: [], DesiredCapabilities: [], Properties: map[]}
11:43:00.656739 RX: Begin{RemoteChannel: 1, NextOutgoingID: 1, IncomingWindow: 5000, OutgoingWindow: 100, HandleMax: 255, OfferedCapabilities: [], DesiredCapabilities: [], Properties: map[]}
11:43:00.656944 TX: Attach{Name: ICnFRFWFOt25iTpacMJ0XgimLfJdZIoUub2J46-h9xVd9ILLOYgjrg, Handle: 0, Role: Sender, SenderSettleMode: unsettled, ReceiverSettleMode: <nil>, Source: <nil>, Target: source{Address: [queue], Durable: 0, ExpiryPolicy: , Timeout: 0, Dynamic: false, DynamicNodeProperties: map[], Capabilities: []}, Unsettled: map[], IncompleteUnsettled: false, InitialDeliveryCount: 0, MaxMessageSize: 0, OfferedCapabilities: [], DesiredCapabilities: [], Properties: map[]}
11:43:01.093420 RX(Session): Attach{Name: ICnFRFWFOt25iTpacMJ0XgimLfJdZIoUub2J46-h9xVd9ILLOYgjrg, Handle: 0, Role: Receiver, SenderSettleMode: <nil>, ReceiverSettleMode: <nil>, Source: <nil>, Target: <nil>, Unsettled: map[], IncompleteUnsettled: false, InitialDeliveryCount: 0, MaxMessageSize: 0, OfferedCapabilities: [], DesiredCapabilities: [], Properties: map[]}
11:43:01.093484 RX(Session): Detach{Handle: 0, Closed: true, Error: *Error{Condition: amqp:unauthorized-access, Description: Unauthorized access. 'Send' claim(s) are required to perform this operation. Resource: 'sb://...'. TrackingId:0598a2a2b50d44fea0bea6fd4aaf81de_G24, SystemTracker:gateway5, Timestamp:2019-06-19T18:43:00, Info: map[]}}
```

**Comment by devigned** [comment]
I believe the issue of the hidden error will be remedied when we bump our amqp lib dependency to `v0.11.1` (see: https://github.com/vcabbage/amqp/pull/162). 

As for the unauthorized access error, it appears the connection string provided doesn't have access to the address `Address: [queue]`. Is `[queue]` something you replaced or is that the actual name of the Hub?

**Comment by metrue** [comment]
So is this issue fixed? I still met this issue today

**Comment by devigned** [comment]
@metrue could you run your code with `DEBUG_LEVEL=3` and `-tags debug`. That should show the amqp protocol level debug.

**Comment by metrue** [comment]
@devigned 

I tried to reproduce that issue but everything works well now,  so I guessed maybe somehow it ran against the old release.

**Comment by metrue** [comment]
it seems like I did not create a policy for it, after I create a policy with`Send` and `Listen`, then initialize an EventHub client with the connection string, everything works well, otherwise it will get error `amqp: sender settlement mode "unsettled" requested, received "mixed" from server` with 404 status code.

**Comment by devigned** [comment]
@gavinfish, I think the AMQP dependency needs to be updated. There was a fix made in 0.11.1 or 0.11.2 which would show the real error. If you have a chance, please update the deps and release a new tag. Ping me if you want to pair.
